lib/runtime/src/pipeline/network/ingress/push_handler.rs

```diff
@@ -220,6 +221,14 @@ where
         let mut send_complete_final = true;
         while let Some(resp) = stream.next().await {
             tracing::trace!("Sending response: {:?}", resp);
+            if let Some(err) = resp.err() {
+                const STREAM_ERR_MSG: &str = "Stream ended before generation completed";
```

Comment:
Just curious - is there any way to get an actual typed error for `resp.err()` propagated here that says it is of type `PyGeneratorExit` or similar, rather than doing string comparison?

Reply:
Yes, I think we should expand on the [MaybeError trait](https://github.com/ai-dynamo/dynamo/blob/b48d4c3bbaf9b5f6b44f544445000c1b6ab303cf/lib/runtime/src/protocols/maybe_error.rs) to have it define a list of standard Dynamo error types, to avoid relying on a predefined string for determining the error type.

Filed a ticket DIS-364